### DISCUSSION TITLE
test problem: MphysAeroStruct: Failed! (Compile on Ubuntu 20.04.4 LTS amd64)

### DISCUSSION THREAD

**Discussion by jh36714753** [mdolab/dafoam#335] [Installation]
Hi, I try to verify the DAFoam installation by running the regression tests: "cd $HOME/dafoam-/repos/dafoam-3.0.0/tests && ./Allrun". But some errors occur:

> --------------------------------------------------------------------------
> MphysAero: Success!
> --------------------------------------------------------------------------
> [[7397,1],1]: A high-performance Open MPI point-to-point messaging module
> was unable to find any relevant network interfaces:
> 
> Module: OpenFabrics (openib)
>   Host: ji-virtual-machine
> 
> Another transport will be used instead, although this may result in
> lower performance.
> 
> NOTE: You can disable this warning by setting the MCA parameter
> btl_base_warn_component_unused to 0.
> --------------------------------------------------------------------------
> Traceback (most recent call last):
>   File "runTests_MphysAeroStruct.py", line 14, in <module>
>     from tacs.mphys import TacsBuilder
> ModuleNotFoundError: No module named 'tacs'
> Traceback (most recent call last):
>   File "runTests_MphysAeroStruct.py", line 14, in <module>
>     from tacs.mphys import TacsBuilder
> ModuleNotFoundError: No module named 'tacs'
> Traceback (most recent call last):
>   File "runTests_MphysAeroStruct.py", line 14, in <module>
>     from tacs.mphys import TacsBuilder
> ModuleNotFoundError: No module named 'tacs'
> Traceback (most recent call last):
>   File "runTests_MphysAeroStruct.py", line 14, in <module>
>     from tacs.mphys import TacsBuilder
> ModuleNotFoundError: No module named 'tacs'
> --------------------------------------------------------------------------
> Primary job  terminated normally, but 1 process returned
> a non-zero exit code. Per user-direction, the job has been aborted.
> --------------------------------------------------------------------------
> --------------------------------------------------------------------------
> mpirun detected that one or more processes exited with non-zero status, thus causing
> the job to be terminated. The first process to do so was:
> 
>   Process name: [[7397,1],2]
>   Exit code:    1
> --------------------------------------------------------------------------
> [ji-virtual-machine:06436] 3 more processes have sent help message help-mpi-btl-base.txt / btl:no-nics
> [ji-virtual-machine:06436] Set MCA parameter "orte_base_help_aggregate" to 0 to see all help / error messages
> MphysAeroStruct: Failed!

I compile from Ubuntu 20.04.4 LTS amd64. I follow the installation guide without any error.

I really hope to get your help.

**Comment by friedenhe** [comment]
Your Tacs installation probably has some problems.

**Reply by jh36714753** [reply]
Thank you, I check my Tacs installation and find some errors:

>        --- Compiled /home/ji/dafoam/repos/tacs/src/io/TACSMarchingCubes.cpp successfully ---
> 
> 
>         --- Compiled /home/ji/dafoam/repos/tacs/src/io/TACSFH5.cpp successfully ---
> 
> 
>         --- Compiled /home/ji/dafoam/repos/tacs/src/io/TACSToFH5.cpp successfully ---
> 
> 
>         --- Compiled /home/ji/dafoam/repos/tacs/src/io/TACSFH5Loader.cpp successfully ---
> 
> 
>         --- Compiled /home/ji/dafoam/repos/tacs/src/io/TACSMeshLoader.cpp successfully ---
> 
> ar rcs /home/ji/dafoam/repos/tacs/lib/libtacs.a /home/ji/dafoam/repos/tacs/src/io/TACSFH5.o /home/ji/dafoam/repos/tacs/src/io/TACSToFH5.o /home/ji/dafoam/repos/tacs/src/io/TACSFH5Loader.o /home/ji/dafoam/repos/tacs/src/io/TACSMeshLoader.o /home/ji/dafoam/repos/tacs/src/io/TACSMarchingCubes.o
> make[1]: Leaving directory '/home/ji/dafoam/repos/tacs/src/io'
> mpicxx -fPIC -shared src/*.o src/bpmat/*.o src/elements/*.o src/elements/dynamics/*.o src/elements/basis/*.o src/elements/shell/*.o src/constitutive/*.o src/functions/*.o src/io/*.o  /home/ji/dafoam/repos/tacs/extern/metis/lib/libmetis.a -L/usr/lib/ -llapack -lpthread -lblas -o /home/ji/dafoam/repos/tacs/lib/libtacs.so
> 
> Usage:   
>   pip install [options] <requirement specifier> [package-index-options] ...
>   pip install [options] -r <requirements file> [package-index-options] ...
>   pip install [options] [-e] <vcs project url> ...
>   pip install [options] [-e] <local project path> ...
>   pip install [options] <archive url/path> ...
> 
> option --use-feature: invalid choice: 'in-tree-build' (choose from '2020-resolver', 'fast-deps', 'truststore')
> make: *** [Makefile:73: interface] Error 2

Can you help me?

**Reply by friedenhe** [reply]
Try to skip `make interface`
